sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -191,11 +193,15 @@ case class SortArray(base: Expression, ascendingOrder: Expression)
 }
 
 /**
- * Checks if the array (left) has the element (right)
+ * Checks if the array (left) has the element (right) and pattern match in
+ * case left is Array of type string
  */
+
 @ExpressionDescription(
-  usage = "_FUNC_(array, value) - Returns TRUE if the array contains the value.",
-  extended = " > SELECT _FUNC_(array(1, 2, 3), 2);\n true")
+  usage = """_FUNC_(array, value) - Returns TRUE if the array contains the value or
```

Comment:
So, in that case we can add one more expression , something like ArrayContainsWithPatternMatch?